### PR TITLE
Remove old resources when yml is updated in custom Apps

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

Remove old resources when yml is updated in custom Apps

### Description of the Change

For custom apps, if developers update their yml file to remove older resources and add new ones, archive the older assets

### CODE REVIEW

File: builds/jazz-build-module/utility-loader.groovy [hunk 1]
```diff
@@ -225,14 +225,25 @@ def constructArn (arn, resourceName, resourceType, config) {
...
+
+def getStackResources (stackName, region, credsId) {
+	def stackResources = null	
+	try {
+		stackResources = sh(script: "aws cloudformation describe-stack-resources --stack-name ${stackName} --region ${region} --profile ${credsId}  --output json 2<&1", returnStdout: true)
+		echo "Describe Stacks are ${stackResources}"
+		def parsedResources = parseJson(stackResources)
+		return parsedResources
+	} catch (ex) {
+		echo "stack not exists..."
```
Comment (raghits) [inline builds/jazz-build-module/utility-loader.groovy]: `stack does not exist`

File: core/jazz_assets/config/global-config.json [hunk 2]
```diff
@@ -81,7 +81,8 @@
 		"provider",
 		"provider_id",
 		"limit",
-		"offset"
+		"offset",
+		"status"
```
Comment (raghits) [inline core/jazz_assets/config/global-config.json]: Any swagger changes needed?

File: core/jazz_assets/components/crud/getList.js [hunk 3]
```diff
@@ -72,22 +86,24 @@ module.exports = (query, asset_table, onComplete) => {
     }
     pagination.offset = query.offset || global.global_config.PAGINATION_DEFAULTS.offset;
 
-    logger.debug("Query params generated from the search request: " + JSON.stringify(params));
+    logger.info("Query params generated from the search request: " + JSON.stringify(params));
```
Comment (raghits) [inline core/jazz_assets/components/crud/getList.js]: `logger.debug`

File: core/jazz_assets/components/crud/getList.js [hunk 4]
```diff
@@ -72,22 +86,24 @@ module.exports = (query, asset_table, onComplete) => {
...
 
     let items = [];
     let queryExecute = (onComplete) => {
         docClient.query(params, (err, data) => {
             let count;
             if (err) {
+                logger.info("GETLIST err: "+ JSON.stringify(err));
```
Comment (raghits) [inline core/jazz_assets/components/crud/getList.js]: `logger.debug`

File: core/jazz_assets/components/crud/getList.js [hunk 5]
```diff
@@ -72,22 +86,24 @@ module.exports = (query, asset_table, onComplete) => {
...
                 onComplete(err, null);
             } else {
+                logger.info("GETLIST data: "+ JSON.stringify(data));
```
Comment (raghits) [inline core/jazz_assets/components/crud/getList.js]: `logger.debug`

File: core/jazz_assets/components/crud/getList.js [hunk 6]
```diff
@@ -46,13 +46,27 @@ module.exports = (query, asset_table, onComplete) => {
     };
 
     let keys_list = global.global_config.ASSET_SEARCH_OPTIONAL_FILTER_PARAMS;
+    logger.info("Query params: " + JSON.stringify(query));
```
Comment (raghits) [inline core/jazz_assets/components/crud/getList.js]: `logger.debug`

File: builds/jenkins-build-pack-sls-app/Jenkinsfile [hunk 7]
```diff
@@ -345,15 +345,21 @@ node() {
...
               def deployOutput = servelessDeploy(environment_logical_id, envBucketKey, credsId)
               // After stack Deployment
 
               createAssetForHttpEvents(deployOutput)
               attachTagstoLogGroup(config, environment_logical_id, credsId)
 
-              def cfStackName = "${instance_prefix}-${config['domain']}-${config['service']}-${environment_logical_id}"
-              def stackResources = utilModule.getStackResources(cfStackName, config['region'], credsId)
-              arnsMap = utilModule.createAllStackResources(whiteListModule, events, config, stackResources, environment_logical_id)
+              def newCloudformationStackResources = utilModule.getStackResources(cfStackName, config['region'], credsId)
```
Comment (raghits) [inline builds/jenkins-build-pack-sls-app/Jenkinsfile]: Can we add comments to clarify what is being done here?

File: builds/jenkins-build-pack-sls-app/Jenkinsfile [hunk 8]
```diff
@@ -345,15 +345,21 @@ node() {
 
               def envBucketKey = "${env_key}"
               echoServerlessFile()
+              def cfStackName = "${instance_prefix}-${config['domain']}-${config['service']}-${environment_logical_id}"
```
Comment (raghits) [inline builds/jenkins-build-pack-sls-app/Jenkinsfile]: Can we add comments to clarify what is being done here?

File: builds/jazz-build-module/utility-loader.groovy [hunk 9]
```diff
@@ -225,14 +225,25 @@ def constructArn (arn, resourceName, resourceType, config) {
...
+		try {
+			def resp = sh(script: "aws cloudformation describe-stack-resources --stack-name ${stackName} --region ${region} --profile ${credsId}  --output json 2<&1 | grep -c 'ValidationError'", returnStdout: true)
+			if(resp != 1) error "describe stack failed."
+			else return stackResources
+		} catch (e) {}
+	}
 }
 
-def createAllStackResources (whiteListModule, events, config, stackResources, env) {
+def createAllStackResources (whiteListModule, events, config, stackResources, env, isSendEvent) {
```
Comment (raghits) [inline builds/jazz-build-module/utility-loader.groovy]: Can we add comments to clarify what is being done here?

Comment (raghits) [reply builds/jazz-build-module/utility-loader.groovy]: `isSendEvent` => `createAsset`

Comment (raghits) [reply builds/jazz-build-module/utility-loader.groovy]: When and why would we not send `CREATE_ASSET`?

File: builds/jazz-build-module/utility-loader.groovy (hunk 1 above)
Comment (SiniWilson) [reply builds/jazz-build-module/utility-loader.groovy]: Done

File: core/jazz_assets/config/global-config.json (hunk 2 above)
Comment (SiniWilson) [reply core/jazz_assets/config/global-config.json]: No. Its already there in swagger,. Here only changing the field as searchable

File: core/jazz_assets/components/crud/getList.js (hunk 3 above)
Comment (SiniWilson) [reply core/jazz_assets/components/crud/getList.js]: Done

File: core/jazz_assets/components/crud/getList.js (hunk 4 above)
Comment (SiniWilson) [reply core/jazz_assets/components/crud/getList.js]: Done

File: core/jazz_assets/components/crud/getList.js (hunk 5 above)
Comment (SiniWilson) [reply core/jazz_assets/components/crud/getList.js]: Done

File: core/jazz_assets/components/crud/getList.js (hunk 6 above)
Comment (SiniWilson) [reply core/jazz_assets/components/crud/getList.js]: Done

File: builds/jenkins-build-pack-sls-app/Jenkinsfile (hunk 7 above)
Comment (SiniWilson) [reply builds/jenkins-build-pack-sls-app/Jenkinsfile]: Done

File: builds/jenkins-build-pack-sls-app/Jenkinsfile (hunk 8 above)
Comment (SiniWilson) [reply builds/jenkins-build-pack-sls-app/Jenkinsfile]: Done

File: builds/jazz-build-module/utility-loader.groovy (hunk 9 above)
Comment (SiniWilson) [reply builds/jazz-build-module/utility-loader.groovy]: Done

Comment (SiniWilson) [reply builds/jazz-build-module/utility-loader.groovy]: We need to send events only after deployment, and we don't need to send the event while we are calling this method only for creating the resource map

**Comment by raghits** [comment]
@devsatishm Can you review this?

**Pull request merged**